quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/PropertyMapper.java

```diff
@@ -234,14 +319,75 @@ public Optional<DeprecatedMetadata> getDeprecatedMetadata() {
         return option.getDeprecatedMetadata();
     }
 
+    /**
+     * An option is considered a wildcard option if its key contains a wildcard placeholder (e.g. log-level-<category>).
+     * The placeholder must be denoted by the '<' and '>' characters.
+     */
+    public boolean hasWildcard() {
+        return fromWildcardPattern != null;
+    }
+
+    /**
+     * Checks if the given option name matches the wildcard pattern of this option.
+     * E.g. check if "log-level-io.quarkus" matches the wildcard pattern "log-level-<category>".
+     */
+    public boolean matchesWildcardOptionName(String name) {
+        if (!hasWildcard()) {
+            return false;
+        }
+        return fromWildcardPattern.matcher(name).matches() || envVarNameWildcardPattern.matcher(name).matches()
+                || (toWildcardPattern != null && toWildcardPattern.matcher(name).matches());
+    }
+
+    /**
+     * Extracts the wildcard value from the given option name.
+     * E.g. for the option "log-level-<category>" and the option name "log-level-io.quarkus",
+     * the wildcard value would be "io.quarkus".
+     */
+    private Optional<String> getWildcardValue(String option, boolean includeMappedToOptions) {
```

Comment:
Is this called with includeMappedToOptions=false?

Also the method / variable names with wildcardValue are a little confusing, because we're actually talking about a key, and in many places the caller is responsible for knowing if the mapper is already a wildcard. Would it be possible to just name this something getKey(originalKey) or mapKey and hide more of the wildcard details in most places?

Reply:
Thanks for the suggestion. I was a little bit strugling with getting the naming right. I refined it a bit. Let me know if it reads better now.